app/views/shops/_form.html.haml

```diff
@@ -25,4 +25,23 @@
 			= f.label :email_id
 			= f.email_field :email_id
 			%br/
-			= f.submit(class: "btn btn-primary")
\ No newline at end of file
+			Location:
+			%input#address{:style => "width: 200px", :type => "text"}/
+			#map{:style => "width: 500px; height: 400px;"}
+			= f.hidden_field :latitude
+			= f.hidden_field :longitude
+			:javascript
```

Comment:
`map.js` file something like this:

```

$(function() {

  var load_map = function() {

    function updateControls(addressComponents) {
      fill_address_field("#parking_space_street", addressComponents.addressLine1);
      fill_address_field("#parking_space_state", addressComponents.stateOrProvince);
      fill_address_field("#parking_space_postal_code", addressComponents.postalCode);
      fill_address_field("#parking_space_city", addressComponents.city);
    }

    var brand_new = (!$("#parking_space_street").val());
    function fill_address_field(element, value) {
      if(!$(element).val() || brand_new) {
        $(element).val(value);
      }
    }

    $('#map').locationpicker({
        // location: {latitude: 53.3482993, longitude: -6.257899},
        location: {latitude: $("#address_lat").val() || 53.3482993, longitude: $('#address_lng').val() || -6.257899},
        inputBinding: {
          latitudeInput: $('#address_lat'),
          longitudeInput: $('#address_lng'),
          locationNameInput: $('#address_lookup')
        },
        enableAutocomplete: true,
        radius: 0,
        zoom: 16,
        scrollwheel: true,
        onchanged: function (currentLocation, radius, isMarkerDropped) {
          var addressComponents = $(this).locationpicker('map').location.addressComponents;
          updateControls(addressComponents);
        }
    });
  }

  if($('#map').length) load_map();

});

```

You need to update all the variable names and IDs in the code above if you are using it exactly. But just use the above as a template for the code you already have.